tck/src/main/java/org/eclipse/microprofile/graphql/tck/dynamic/GraphQLDynamicClientTest.java

```diff
@@ -0,0 +1,262 @@
+import org.json.JSONException;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import lombok.extern.java.Log;
+
+/**
+ * This test runs all test defined in 'src/test/resources'
+ * @author Phillip Kruger (phillip.kruger@redhat.com)
+ */
+@Log
+public class GraphQLDynamicClientTest extends Arquillian {
+    private static class GetSysPropAction implements PrivilegedAction<String> {
+        private final String propertyName;
+        private final String defaultValue;
+
+        GetSysPropAction(String propertyName, String defaultValue) {
+            this.propertyName = propertyName;
+            this.defaultValue = defaultValue;
+        }
+
+        @Override
+        public String run() {
+            return System.getProperty(propertyName, defaultValue);
+        }
+        
+    }
+
+    private static final String UTF8 = "utf-8";
+    private static final String MEDIATYPE_JSON = "application/json";
+    private static final String HEADER_CONTENT_TYPE = "Content-Type";
+    private static final String HEADER_ACCEPT = "Accept";
+    private static final String QUERY = "query";
+    private static final String VARIABLES = "variables";
+    private static final String ENDPOINT_URL = "http://localhost:" + 
+        AccessController.doPrivileged(new GetSysPropAction("mp.graphql.tck.endpoint.port", "8080")) + "/SuperHeroDatabase/graphql";
+    private static final String CONNECT_TIMEOUT = AccessController.doPrivileged(new GetSysPropAction("mp.graphql.tck.connect.timeout", "30000"));
+    private static final String READ_TIMEOUT = AccessController.doPrivileged(new GetSysPropAction("mp.graphql.tck.read.timeout", "30000"));
+    private static File testDataDir;
+    private URL graphqlEndpoint;
+
+    @Deployment
+    public static Archive<?> createDeployment() throws Exception {
+        testDataDir = GraphQLTestDataProvider.setupTestDataDir();
+        Archive<?> archive = ShrinkWrap.create(WebArchive.class, "SuperHeroDatabase.war")
+            .addPackage(HeroFinder.class.getPackage())
+            .addPackage(HeroDatabase.class.getPackage())
+            .addPackage(SuperHero.class.getPackage())
+            .addPackage(GraphQLDynamicClientTest.class.getPackage())
+            .addPackage(TestData.class.getPackage())
+            .add(new StringAsset(testDataDir.getAbsolutePath()), "testDataDir.txt")
+            .addAsLibrary(new File(IOUtils.class.getProtectionDomain().getCodeSource().getLocation().toURI()))
+            .addAsLibrary(new File(JSONException.class.getProtectionDomain().getCodeSource().getLocation().toURI()))
+            .addAsLibrary(new File(JSONAssert.class.getProtectionDomain().getCodeSource().getLocation().toURI()));
+            
+            archive.as(ZipExporter.class).exportTo(new File("/Users/andymc/tmp/" + archive.getName()), true);
```

Comment:
This might be a problem ? Maybe `System.getProperty("java.io.tmpdir")` ?

Reply:
Good catch - I meant to remove that before I pushed.  It still could be useful for debug - do you have a preference between removing that line vs using java.io.tmpdir?

Reply:
Ok so that is only for debugging ? For now java.io.tmpdir but maybe we remove it later ?